pandas/core/dtypes/cast.py

```diff
@@ -1559,6 +1559,11 @@ def construct_1d_arraylike_from_scalar(
             dtype = np.dtype("object")
             if not isna(value):
                 value = ensure_str(value)
+        elif isinstance(dtype, np.dtype) and dtype.kind == "M" and value is NaT:
```

Comment:
use is_nat_for_dtype (is not exactly this same but similar)

Reply:
is_valid_nat_for_dtype is the function

Reply:
I couldn't just use `is_valid_nat_for_dtype` as it also returned true for `is_valid_nat_for_dtype(nan, float64)` (which was a problem for the `test_modulo2` test case for instance (I didn't check others which also failed)).